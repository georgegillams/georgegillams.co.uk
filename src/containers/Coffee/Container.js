import React from 'react';
import PageTitle from 'components/common/PageTitle';

import PageContainer from 'components/common/PageContainer';
import Paragraph from '@george-gillams/components/paragraph';
import TextLink from '@george-gillams/components/text-link';
import { VStack } from 'components/common/Stacks';
import { spacingBase } from '@george-gillams/components/constants/layout';

const links = [
  { url: 'https://donation.dec.org.uk/ukraine-humanitarian-appeal', name: 'DEC Ukraine Humanitarian Appeal' },
  { url: 'https://www.redcross.org.uk/', name: 'Red Cross' },
  { url: 'https://www.beateatingdisorders.org.uk/', name: 'Beat' },
];

const Coffee = () => {
  return (
    <PageContainer bottomPadding>
      <PageTitle anchor={false} name="Buy me a coffee"></PageTitle>
      <Paragraph>
        Thanks for thinking of me and considering sending some cash my way. As a developer of free apps, I really
        appreciate people who want to pay me for my work.
      </Paragraph>
      <Paragraph>
        The world is kinda on fire right now though, so if you have spare cash, please consider sending it to one of
        these great causes instead:
      </Paragraph>
      <VStack style={{ marginTop: spacingBase }}>
        {links.map(link => (
          <TextLink href={link.url} hrefExternal key={link.url}>
            {link.name}
          </TextLink>
        ))}
      </VStack>
    </PageContainer>
  );
};

export default Coffee;
