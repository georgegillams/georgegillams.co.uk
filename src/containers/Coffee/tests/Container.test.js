import React from 'react';
import { render } from '@testing-library/react';

import Coffee from '../Container';

describe('<Coffee />', () => {
  it('should render correctly', () => {
    const { container } = render(<Coffee />);

    expect(container).toMatchSnapshot();
  });
});
