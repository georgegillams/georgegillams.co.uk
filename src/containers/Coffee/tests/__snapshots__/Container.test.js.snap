// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Coffee /> should render correctly 1`] = `
.c10 {
  width: 1.25rem;
  height: 1.25rem;
}

.c7 {
  display: inline-block;
  border-radius: 0.25rem;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  color: #9932cc;
  fill: #9932cc;
}

.c7:focus-within,
.c7:focus {
  position: relative;
  outline: none !important;
}

.c7:focus-within::after,
.c7:focus::after {
  position: absolute;
  top: -0.35rem;
  right: -0.35rem;
  bottom: -0.35rem;
  left: -0.35rem;
  content: '';
  display: block;
  border: solid #007dfa3C 0.2rem;
  border-radius: 0.5rem;
  pointer-events: none;
}

.c7:hover {
  color: #44aeff;
  fill: #44aeff;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 0.4rem;
  height: 1rem;
  margin-left: 0.55rem;
}

.c11 {
  position: absolute;
  width: 1rem;
  height: 1rem;
  margin-top: 0.12rem;
  margin-left: -0.45rem;
}

.c3 {
  margin: 0;
  font-size: 2.4rem;
  font-weight: bold;
}

.c8 {
  margin: 0;
}

.c4 {
  margin: 2.3rem 0 0 0;
}

.c1 {
  padding-top: 0;
  opacity: 1;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.c2 {
  -webkit-animation: bbwGtI 0.4s ease-in-out;
  animation: bbwGtI 0.4s ease-in-out;
}

.c0 {
  display: block;
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  overflow: hidden;
  padding-bottom: 2rem;
  max-width: 60rem;
  padding-right: 1rem;
  padding-left: 1rem;
}

.c5 {
  display: inline-block;
  margin-bottom: 0;
  padding-bottom: 0;
  line-height: 1.5;
  margin-top: 2rem;
}

.c5:first-of-type {
  margin-top: 0.86rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 > *:not(:first-child) {
  margin-top: 1rem;
}

@media (prefers-color-scheme:dark) {
  .c7:focus-within::after,
  .c7:focus::after {
    border-color: #007dfa3C;
  }
}

@media (prefers-color-scheme:dark) {
  .c7 {
    color: #da70d6;
    fill: #da70d6;
  }

  .c7:hover {
    color: #44aeff;
    fill: #44aeff;
  }
}

@media (min-width:50.25rem) {
  .c0 {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}

<div>
  <div
    class="c0 page-containerstyles__StyledPageContainer-sc-19l7k4-0"
    width="default"
  >
    <div>
      <style>
        .js .animated-content__wrapper--hidden {
  opacity: 0.01;
}
      </style>
      <div
        class="c1 animated-content__wrapper--hidden"
      >
        <div
          class="c2"
        >
          <h1
            class="c3 c4"
          >
            Buy me a coffee
          </h1>
        </div>
        <span
          class="c5"
        >
          Thanks for thinking of me and considering sending some cash my way. As a developer of free apps, I really appreciate people who want to pay me for my work.
        </span>
        <span
          class="c5"
        >
          The world is kinda on fire right now though, so if you have spare cash, please consider sending it to one of these great causes instead:
        </span>
        <div
          class="c6"
          style="margin-top: 1rem;"
        >
          <a
            aria-label="DEC Ukraine Humanitarian Appeal (opens in a new tab)"
            class="c7"
            href="https://donation.dec.org.uk/ukraine-humanitarian-appeal"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="c8"
            >
              DEC Ukraine Humanitarian Appeal
            </span>
            <div
              class="c9"
            >
              <svg
                class="c10 c11"
                height="18"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.5 11.3c.4.4 1 .4 1.4 0L19 6.2V9c0 .6.4 1 1 1s1-.4 1-1V4c0-.6-.4-1-1-1h-5c-.6 0-1 .4-1 1s.4 1 1 1h2.4l-4.9 4.9c-.4.4-.4 1 0 1.4zM19 19c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V7c0-1.1.9-2 2-2h7v2H5v12h12v-7h2"
                />
              </svg>
            </div>
          </a>
          <a
            aria-label="Red Cross (opens in a new tab)"
            class="c7"
            href="https://www.redcross.org.uk/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="c8"
            >
              Red Cross
            </span>
            <div
              class="c9"
            >
              <svg
                class="c10 c11"
                height="18"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.5 11.3c.4.4 1 .4 1.4 0L19 6.2V9c0 .6.4 1 1 1s1-.4 1-1V4c0-.6-.4-1-1-1h-5c-.6 0-1 .4-1 1s.4 1 1 1h2.4l-4.9 4.9c-.4.4-.4 1 0 1.4zM19 19c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V7c0-1.1.9-2 2-2h7v2H5v12h12v-7h2"
                />
              </svg>
            </div>
          </a>
          <a
            aria-label="Beat (opens in a new tab)"
            class="c7"
            href="https://www.beateatingdisorders.org.uk/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="c8"
            >
              Beat
            </span>
            <div
              class="c9"
            >
              <svg
                class="c10 c11"
                height="18"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.5 11.3c.4.4 1 .4 1.4 0L19 6.2V9c0 .6.4 1 1 1s1-.4 1-1V4c0-.6-.4-1-1-1h-5c-.6 0-1 .4-1 1s.4 1 1 1h2.4l-4.9 4.9c-.4.4-.4 1 0 1.4zM19 19c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V7c0-1.1.9-2 2-2h7v2H5v12h12v-7h2"
                />
              </svg>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
